gc/base/MarkingScheme.cpp

```diff
@@ -272,14 +294,15 @@ MM_MarkingScheme::setMarkBitsInRange(MM_EnvironmentBase *env, void *heapBase, vo
 void
 MM_MarkingScheme::completeScan(MM_EnvironmentBase *env)
 {
-	omrobjectptr_t objectPtr = NULL;
-	MM_WorkPackets *packets = getWorkPackets();
-
+	uintptr_t *bytesScanned = &(env->_markStats._bytesScanned);
```

Comment:
This (completeScan()) is the innermost scanning loop. I also factored out of scanObject() a more efficient private version of scanObject() that avoids duplicate NULL check on objects returned by scanners. When I did that it just made sense to move stats update into completeScan() call and cache them. I agree that, with this refactoring of stats update out of reduced scanObject(), smart compiler should cache stats addresses anyway, but not so sure that it would do so if stats were always updated in scanObject().

The caching is done only the private scanObject(), and takes away any question of whether the compiler is smart enough to do this. I don't agree that this caching makes the code more complex. And I think that the private scanObject() has an overall positive impact on marking scan times.